firstvoices/backend/views/search/base_search_views.py

```diff
@@ -57,15 +95,24 @@ def get_search_params(self):
         """
         input_q = self.request.GET.get("q", "")
 
-        search_params = {"q": input_q, "site_slug": ""}
+        input_types_str = self.request.GET.get("types", "")
+        valid_types_list = get_valid_document_types(input_types_str)
+
+        search_params = {"q": input_q, "types": valid_types_list, "site_slug": ""}
         return search_params
 
     def list(self, request, **kwargs):
         search_params = self.get_search_params()
 
+        # If no valid types are passed, return emtpy list as a response
+        if not search_params["types"]:
+            return Response(data=[])
```

Comment:
Is this correct? If they don't specify types, I would expect all data types to be returned.

Reply:
so due to the default behaviour of get_search_params function, if a user does not specify types, it will add all types to search_params["types"].
The above case will only be triggered, when the user has specified all invalid types, so the get_search_params function makes `search_params["types"]=None` which will return empty results.